osgp/platform/osgp-adapter-domain-publiclighting/src/main/java/org/opensmartgridplatform/adapter/domain/publiclighting/application/tasks/EventRetrievalScheduledTask.java

```diff
@@ -58,15 +66,75 @@ public void run() {
                 return;
             }
 
-            final List<Device> devicesToContact = this.findDevicesToContact(devices,
+            List<Device> devicesToContact = this.findDevicesToContact(devices,
                     this.eventRetrievalScheduledTaskMaximumAllowedAge);
             if (devicesToContact == null || devicesToContact.isEmpty()) {
                 return;
             }
 
-            this.contactDevices(devicesToContact, DeviceFunction.GET_STATUS);
+            devicesToContact = this.filterByExponentialBackOff(devicesToContact);
+
+            this.contactDevices(devicesToContact, DeviceFunction.GET_LIGHT_STATUS);
         } catch (final Exception e) {
             LOGGER.error("Exception caught during EventRetrievalScheduledTask.run()", e);
         }
     }
+
+    /**
+     * Using the connection details of a device, determine if the device should
+     * be contacted. Devices without connection error included in the returned
+     * list. Device with connection errors will be included in the returned list
+     * if the exponential back off period has elapsed.
+     *
+     * @param devicesToFilter
+     *            List of devices.
+     *
+     * @return Filtered list of devices based on the exponential back off
+     *         calculations.
+     */
+    private List<Device> filterByExponentialBackOff(final List<Device> devicesToFilter) {
+
+        final List<Device> devicesToContact = new ArrayList<>();
+
+        for (final Device device : devicesToFilter) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS135)